x/wasm-storage/keeper/msg_server.go

```diff
@@ -56,19 +59,38 @@ func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 
 var _ types.MsgServer = msgServer{}
 
+// StoreDataRequestWasm handles the processing of a store data request for a wasm-storage module.
+// It unzips the Wasm bytecode, and creates a new Wasm instance,
+// checks if a module with the same hash already exists, sets the Wasm instance in the data store,
+// tracks its expiration, emits an event, and returns the hash of the stored Wasm module.
 func (m msgServer) StoreDataRequestWasm(goCtx context.Context, msg *types.MsgStoreDataRequestWasm) (*types.MsgStoreDataRequestWasmResponse, error) {
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
 	ctx := sdk.UnwrapSDKContext(goCtx)
+	params, err := m.Params.Get(ctx)
+	if err != nil {
+		return nil, err
+	}
 
-	unzipped, err := unzipWasm(msg.Wasm)
+	unzipped, err := unzipWasm(msg.Wasm, params.MaxWasmSize)
 	if err != nil {
 		return nil, err
 	}
-	wasm := types.NewWasm(unzipped, msg.WasmType, ctx.BlockTime())
+
+	wasm := types.NewWasm(unzipped, msg.WasmType, ctx.BlockTime(), ctx.BlockHeight(), params.WasmTTL)
```

Comment:
Perhaps `NewWasm` should return a pointer so that we don't have to dereference it?

Reply:
Addressed here: [a1f2999](https://github.com/sedaprotocol/seda-chain/pull/266/commits/a1f29999f6d95815f78110a6d6c910f6f3359614)